src/lib/optimizer/strategy/projection_combination_rule.cpp

```diff
@@ -0,0 +1,92 @@
+#include "projection_combination_rule.hpp"
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "logical_query_plan/abstract_lqp_node.hpp"
+#include "logical_query_plan/projection_node.hpp"
+
+namespace opossum {
+
+std::string ProjectionCombinationRule::name() const { return "Projection Combination Rule"; }
+
+bool ProjectionCombinationRule::apply_to(const std::shared_ptr<AbstractLQPNode>& node) {
+  if (node->type() != LQPNodeType::Projection) {
+    return _apply_to_children(node);
+  }
+
+  std::vector<std::shared_ptr<ProjectionNode>> projection_nodes;
+
+  // Gather subsequent ProjectionNodes
+  auto current_node = node;
+  while (current_node->type() == LQPNodeType::Projection) {
+    projection_nodes.emplace_back(std::static_pointer_cast<ProjectionNode>(current_node));
+
+    DebugAssert(current_node->left_child(), "LQP invalid: ProjectionNode must have a left child.");
+    DebugAssert(!current_node->right_child(), "LQP invalid: ProjectionNode can not have a right child.");
+
+    current_node = current_node->left_child();
+
+    if (current_node->parents().size() > 1) {
+      break;
+    }
+  }
+
+  if (projection_nodes.size() > 1) {
+    const auto combined_projection_node = _combine_projection_nodes(projection_nodes);
+    _apply_to_children(combined_projection_node);
+    return true;
+  }
+
+  return _apply_to_children(node);
+}
+
+std::shared_ptr<ProjectionNode> ProjectionCombinationRule::_combine_projection_nodes(
+    std::vector<std::shared_ptr<ProjectionNode>>& projection_nodes) const {
+  // Store original child and parents
+  auto left_child = projection_nodes.back()->left_child();
+  const auto parents = projection_nodes.front()->parents();
+  const auto child_sides = projection_nodes.front()->get_child_sides();
+
+  auto column_expressions = std::vector<std::shared_ptr<LQPExpression>>();
+  for (const auto expression : projection_nodes.front()->column_expressions()) {
+    // If the expression is no reference to another column, we can just add it.
+    if (expression->type() != ExpressionType::Column) {
+      column_expressions.push_back(expression);
```

Comment:
If this Expression is a nested Expression and references any of the ProjectionNodes you are removing, then we would have invalid LQPColumnReferences.